src/test/java/seedu/address/testutil/DebugModelManager.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.testutil;
+
+import seedu.address.model.ModelManager;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.ReadOnlyCalendar;
+import seedu.address.model.UserPrefs;
+
+/**
+ * Debug Model Manager that prints comparison discrepancies
+ */
+public class DebugModelManager extends ModelManager {
+    public DebugModelManager(ReadOnlyAddressBook addressBook, ReadOnlyCalendar calendar, UserPrefs userPrefs) {
+        super(addressBook, calendar, userPrefs);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        // short circuit if same object
+        if (obj == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(obj instanceof DebugModelManager)) {
+            return false;
+        }
+
+        // state check
+        DebugModelManager other = (DebugModelManager) obj;
+        if (versionedAddressBook.equals(other.versionedAddressBook)) {
+            if (filteredStudents.equals(other.filteredStudents)) {
+                if (versionedCalendar.equals(other.versionedCalendar)) {
+                    if (filteredEvents.equals(other.filteredEvents)) {
+                        return true;
+                    }
+                    System.out.println("Filtered events don't match");
```

Comment:
Would there be an advantage to using this as opposed to the equals method in the ModelManager itself? This class itself seems redundant. Also, if you wanted to debug, logging would probably be better.

Reply:
removed DebugModelManager